support-services/src/test/scala/com/gu/support/redemption/RedemptionSpec.scala

```diff
@@ -0,0 +1,79 @@
+package com.gu.support.redemption
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue
+import com.gu.support.redemption.GetCodeStatus.{CodeAlreadyUsed, NoSuchCode}
+import com.gu.support.redemption.Redemption.RedemptionCode
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.Future
+
+class RedemptionSpec extends AsyncFlatSpec with Matchers {
+
+  "getCodeStatus" should "handle an available code" in {
+    val getCodeStatus = GetCodeStatus.withDynamoLookup {
+      case "CODE" => Future.successful(Some(Map("available" -> new AttributeValue().withBOOL(true))))
+    }
+    getCodeStatus(RedemptionCode("CODE")).map {
+      _ should be(Right(()))
+    }
+  }
```

Comment:
if we wanted to make the test really neat, we could get parameterise `GetCodeStatus` on the `Future` and pass in `Id`.  This would get rid of the Future.successful and the .map, but I think the overall effect on the code is negative, and I'm happy enough with the clarity of the tests without.